### ISSUE TITLE
null builder does not accept winrm as a communicator

### ISSUE THREAD

**Issue by NicCOConnor** [mitchellh/packer#2439]
Using a null builder with a winrm communicator gives an error stating that host, ssh_username, and ssh_password are not defined. Packer should be checking for winrm_host, winrm_username, winrm_password instead. The result was realized with ``` packer validate example.json ``` 
builders portion of code

```json
  "builders": [
    {
      "type": "null",
      "communicator": "winrm",
      "winrm_host": "<redacted>",
      "winrm_port": "9585",
      "winrm_username": "Administrator",
      "winrm_password": "<redacted>"
    }
  ],
```
output 
```
PS D:\Users\Nic\Source\packer-windows> packer build .\windows_2008_r2-debug.json
null output will be in this color.

3 error(s) occurred:

* host must be specified
* ssh_username must be specified
* one of ssh_password and ssh_private_key_file must be specified
```

**Comment by cbednarski** [comment]
@NicCOConnor Thanks for the report. I think winrm was added just after the null provisioner and we haven't updated it to support winrm yet.

**Comment by pecigonzalo** [comment]
+1

**Comment by dhoer** [comment]
+1

**Comment by pecigonzalo** [comment]
Should be closed now on #2525
